swift/swift.py

```diff
@@ -204,50 +227,58 @@ def _swiftcall(self, msg: str):
               "destory": Destroy an app.
                 Its "args" is a dictionary with a key; "name".
                 The value of "name" is a name of app you want to destroy.
+        
+        Raises:
+            RuntimeError: When the user rejects the request.
+        
+        Returns:
+            The returned value of the swift-call, if any.
         """
-        try:
-            msg = json.loads(msg)
-        except json.JSONDecodeError as e:
-            print(f"swift.swift._swiftcall(): {e!r}")
-            return
-        if any(key not in msg for key in ("action", "args")):
-            print("The message was ignored because "
-                  "it has no such key; action or args.")
-            return
+        msg = json.loads(msg)
         action, args = msg["action"], msg["args"]
         if action == "create":
-            if any(key not in args for key in ("name", "info")):
-                print("The message was ignored because "
-                      "args of the create action have no such key; name or info.")
-                return
             name, info = args["name"], args["info"]
             reply = QMessageBox.warning(
                 None,
                 "swift-call",
-                f"The app {self.sender().name} requests for creating an app {name}",
+                f"The app {sender} requests for creating an app {name}",
                 QMessageBox.Ok | QMessageBox.Cancel,
                 QMessageBox.Cancel
             )
             if reply == QMessageBox.Ok:
-                self.createApp(name, AppInfo(**info))
-        elif action == "destroy":
-            if any(key not in args for key in ("name",)):
-                print("The message was ignored because "
-                      "args of the destroy action have no such key; name.")
-                return
+                return self.createApp(name, AppInfo(**info))
+            raise RuntimeError("user rejected the request.")
+        if action == "destroy":
             name = args["name"]
             reply = QMessageBox.warning(
                 None,
                 "swift-call",
-                f"The app {self.sender().name} requests for destroying an app {name}",
+                f"The app {sender} requests for destroying an app {name}",
                 QMessageBox.Ok | QMessageBox.Cancel,
                 QMessageBox.Cancel
             )
             if reply == QMessageBox.Ok:
-                self.destroyApp(name)
+                return self.destroyApp(name)
+            raise RuntimeError("user rejected the request.")
+        raise NotImplementedError(action)
+
+    def _swiftcall(self, sender: str, msg: str):

```

Comment:
Why don't you use `@pyqtSlot`?

I'm not sure, but if you use it, `sender` may be not necessary since it can be obtained by `self.sender()`.

Reply:
As I said in the initial comment of this PR:

> 2. [QObject.sender()](https://doc.qt.io/qt-5/qobject.html#sender) is convenient, but we have to be careful as the document says, which implies that it easily breaks our assumptions and intuition. For example, consider the case where a slot is not called by a signal.

I would prefer not to use `self.sender()` if possible.